src/Data/Nat/Binary/Minus.agda

```diff
@@ -0,0 +1,239 @@
+odd∸even-for> {x} {y} x>y  with minus x y
+... | (_ , tri< x<y _   _) =  contradiction x>y (<⇒≯ x<y)
+... | (_ , tri≈ _   x≡y _) =  contradiction x≡y (>⇒≢ x>y)
+... | (_ , tri> _   _   _) =  refl
+
+x≤y⇒x∸y≡0 :  ∀ {x y} → x ≤ y → x ∸ y ≡ 0ᵇ
+x≤y⇒x∸y≡0 {zero} {y} _                       =  0∸x≡0 y
+x≤y⇒x∸y≡0 (inj₁ (even<even x<y))             =  x≡0⇒double[x]≡0 (x≤y⇒x∸y≡0 (inj₁ x<y))
+x≤y⇒x∸y≡0 {2[1+ x ]} {2[1+ .x ]} (inj₂ refl) =  x≡0⇒double[x]≡0 (x≤y⇒x∸y≡0 {x} {x} (inj₂ refl))
+x≤y⇒x∸y≡0 {2[1+ x ]} {1+[2 y ]} (inj₁ (even<odd x<y))  with minus x y
+... | (0ᵇ , tri< _ _   _  ) =  refl
+... | (_ ,  tri≈ _ x≡y _  ) =  contradiction x≡y (<⇒≢ x<y)
+... | (_ ,  tri> _ _   x>y) =  contradiction x>y (<⇒≯ x<y)
+
+x≤y⇒x∸y≡0 {1+[2 x ]} {2[1+ y ]} (inj₁ (odd<even x≤y))  with minus x y
+... | (_ , tri< _ _ _  ) =  refl
+... | (_ , tri≈ _ _ _  ) =  refl
+... | (_ , tri> _ _ x>y) =  contradiction x>y (≤⇒≯ x≤y)
+
+x≤y⇒x∸y≡0 {1+[2 _ ]} {2[1+ _ ]} (inj₂ ())
+
+x≤y⇒x∸y≡0 {1+[2 _ ]} {1+[2 _ ]}  (inj₁ (odd<odd x<y)) =  x≡0⇒double[x]≡0 (x≤y⇒x∸y≡0 (inj₁ x<y))
+x≤y⇒x∸y≡0 {1+[2 x ]} {1+[2 .x ]} (inj₂ refl)          =
+  x≡0⇒double[x]≡0 (x≤y⇒x∸y≡0 {x} {x} (inj₂ refl))
+
+x∸x≡0 :  ∀ x → x ∸ x ≡ 0ᵇ
+x∸x≡0 x =  x≤y⇒x∸y≡0 (≤-refl {x})
+
+y<x⇒x∸y≢0 :  ∀ {x y} → y < x → x ∸ y ≢ 0ᵇ
+y<x⇒x∸y≢0 {zero} {_} ()
+y<x⇒x∸y≢0 {x'@(2[1+ x ])} {zero} _ = subst (_≢ 0ᵇ) (sym x'∸0≡x') 2[1+x]≢0  where
+                                                                           x'∸0≡x' = x∸0≡x x'
+y<x⇒x∸y≢0 {2[1+ _ ]} {2[1+ _ ]} (even<even y<x)  =  x≢0⇒double[x]≢0 (y<x⇒x∸y≢0 y<x)
+y<x⇒x∸y≢0 {2[1+ x ]} {1+[2 y ]} (odd<even y≤x)   with minus x y
+... | (_ , tri< x<y _ _) =  contradiction y≤x (<⇒≱ x<y)
+... | (d , tri≈ _   _ _) =  1+[2x]≢0 {d}
+... | (d , tri> _   _ _) =  1+[2x]≢0 {d}
+y<x⇒x∸y≢0 {x'@(1+[2 x ])} {y'@(2[1+ y ])} (even<odd y<x) =
+  subst (_≢ 0ᵇ) (sym x'∸y'≡pred[2d]) pred[2d]≢0
+  where
+  -- d = x ∸ y
+  d≢0        = y<x⇒x∸y≢0 y<x;                x'∸y'≡pred[2d] = odd∸even-for> y<x
+  1≤d        = x<y⇒suc[x]≤y (x≢0⇒x>0 d≢0);   2≤2d           = double-mono-≤ 1≤d
+  1≤pred[2d] = pred-mono-≤ 2≤2d;             0<pred[2d]     = suc[x]≤y⇒x<y 1≤pred[2d]
+  pred[2d]≢0 = >⇒≢ 0<pred[2d]
+
+y<x⇒x∸y≢0 {1+[2 _ ]} {1+[2 _ ]} (odd<odd y<x) =  x≢0⇒double[x]≢0 (y<x⇒x∸y≢0 y<x)
+
+x∸y≡0⇒x≤y :  ∀ {x y} → x ∸ y ≡ 0ᵇ → x ≤ y
+x∸y≡0⇒x≤y {x} {y} x∸y≡0  with x ≤? y
+... | yes leq = leq
+... | no x≰y  = contradiction x∸y≡0 (y<x⇒x∸y≢0 (≰⇒> x≰y))
+
+x<y⇒0<y∸x :  ∀ {x y} → x < y → 0ᵇ < y ∸ x
+x<y⇒0<y∸x {x} {y} x<y  with <-cmp (y ∸ x) 0ᵇ
+... | tri> _ _     gt =  gt
+... | tri≈ _ y∸x≡0 _  =  contradiction (x∸y≡0⇒x≤y y∸x≡0) (<⇒≱ x<y)
+... | tri< y∸x<0 _ _  =  contradiction y∸x<0 x≮0
+
+y≤x⇒x≡[x∸y]+y :  ∀ {x y} → y ≤ x → x ≡ (x ∸ y) + y    -- the defining property for _∸_
+y≤x⇒x≡[x∸y]+y {zero} {y} y≤0 =  sym $ begin-equality
+  (0ᵇ ∸ y) + y    ≡⟨ cong₂ _+_ (cong (0ᵇ ∸_) y≡0) y≡0 ⟩
+  (0ᵇ ∸ 0ᵇ) + 0ᵇ  ≡⟨ +-identityʳ 0ᵇ ⟩
+  0ᵇ              ∎
+  where open IneqReasoning;  y≡0 = x≤0⇒x≡0 y≤0
```

Comment:
You can just open `≤-Reasoning` once at the top of the file.

Reply:
In the coming Binary.Minus, it is also used ≡-Reasoning for Nat (not fir Bin).
So, to avoid a name clash, I am forced to open the two Reasonings locally.
(?)